url7f729337e3_timmolter_Sundial_tgz-pJ8-org_knowm_sundial_SampleRunJ8/src/org/quartz/core/SimpleThreadPool.java

```diff
@@ -0,0 +1,567 @@
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Constructors.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  /**
+   * <p>
+   * Create a new (unconfigured) <code>SimpleThreadPool</code>.
+   * </p>
+   * 
+   * @see #setThreadCount(int)
+   * @see #setThreadPriority(int)
+   */
+  public SimpleThreadPool() {
+
+  }
+
+  /*
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Interface.
+   * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   */
+
+  public Logger getLog() {
+
+    return log;
+  }
+
+  @Override
+  public int getPoolSize() {
+
+    return getThreadCount();
+  }
+
+  /**
+   * <p>
+   * Set the number of worker threads in the pool - has no effect after <code>initialize()</code> has been called.
+   * </p>
+   */
+  public void setThreadCount(int count) {
+
+    this.count = count;
+  }
+
+  /**
+   * <p>
+   * Get the number of worker threads in the pool.
+   * </p>
+   */
+  public int getThreadCount() {
+
+    return count;
+  }
+
+  /**
+   * <p>
+   * Set the thread priority of worker threads in the pool - has no effect after <code>initialize()</code> has been called.
+   * </p>
+   */
+  public void setThreadPriority(int prio) {
+
+    this.prio = prio;
+  }
+
+  /**
+   * <p>
+   * Get the thread priority of worker threads in the pool.
+   * </p>
+   */
+  public int getThreadPriority() {
+
+    return prio;
+  }
+
+  public void setThreadNamePrefix(String prfx) {
+
+    this.threadNamePrefix = prfx;
+  }
+
+  public String getThreadNamePrefix() {
+
+    if (threadNamePrefix == null) {
+      threadNamePrefix = schedulerInstanceName + "-SimpleThreadPoolWorker";
+    }
+    return threadNamePrefix;
+  }
+
+  /**
+   * @return Returns the threadsInheritContextClassLoaderOfInitializingThread.
+   */
+  public boolean isThreadsInheritContextClassLoaderOfInitializingThread() {
+
+    return inheritLoader;
+  }
+
+  /**
+   * @param inheritLoader The threadsInheritContextClassLoaderOfInitializingThread to set.
+   */
+  public void setThreadsInheritContextClassLoaderOfInitializingThread(boolean inheritLoader) {
+
+    this.inheritLoader = inheritLoader;
+  }
+
+  public boolean isThreadsInheritGroupOfInitializingThread() {
+
+    return inheritGroup;
+  }
+
+  public void setThreadsInheritGroupOfInitializingThread(boolean inheritGroup) {
+
+    this.inheritGroup = inheritGroup;
+  }
+
+  /**
+   * @return Returns the value of makeThreadsDaemons.
+   */
+  public boolean isMakeThreadsDaemons() {
+
+    return makeThreadsDaemons;
+  }
+
+  /**
+   * @param makeThreadsDaemons The value of makeThreadsDaemons to set.
+   */
+  public void setMakeThreadsDaemons(boolean makeThreadsDaemons) {
+
+    this.makeThreadsDaemons = makeThreadsDaemons;
+  }
+
+  public void setInstanceId(String schedInstId) {
+
+  }
+
+  @Override
+  public void initialize() throws SchedulerConfigException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 216-249 into a separate method.